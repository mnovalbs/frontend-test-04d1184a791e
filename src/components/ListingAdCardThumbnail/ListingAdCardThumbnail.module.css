.thumbnailContainer {
  position: relative;
  width: 100%;
  padding-top: 50%;
  background-color: #eaeaea;
}

.thumbnail {
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  object-fit: cover;
}