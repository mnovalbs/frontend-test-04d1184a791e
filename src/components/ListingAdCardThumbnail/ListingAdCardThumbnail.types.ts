export interface ListingAdCardThumbnailProps {
  alt: string;
  src: string;
}
