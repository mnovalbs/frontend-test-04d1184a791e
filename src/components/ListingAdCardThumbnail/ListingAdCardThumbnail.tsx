import { createThumbnailUrls } from "utils/string";
import styles from "./ListingAdCardThumbnail.module.css"
import { ListingAdCardThumbnailProps } from "./ListingAdCardThumbnail.types";

function ListingAdCardThumbnail({ alt, src }: ListingAdCardThumbnailProps) {
  const [imageUrl1x, imageUrl2x] = createThumbnailUrls(src, 544);

  return (
    <div className={styles.thumbnailContainer}>
      <img
        className={styles.thumbnail}
        alt={alt}
        loading="lazy"
        width="544"
        height="272"
        src={imageUrl1x}
        srcSet={`${imageUrl1x}, ${imageUrl2x} 2x`}
      />
    </div>
  );
}

export default ListingAdCardThumbnail;
