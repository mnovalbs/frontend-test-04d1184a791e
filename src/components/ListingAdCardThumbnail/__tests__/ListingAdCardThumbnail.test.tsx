import { render } from "@testing-library/react"
import mockAd from "contants/mockAd"
import ListingAdCardThumbnail from "../ListingAdCardThumbnail"

describe("ListingAdCardThumbnail", () => {
  it('renders thumbnail', () => {
    const { container } = render(<ListingAdCardThumbnail alt={mockAd.title} src={mockAd.pic} />)

    expect(container).toMatchSnapshot()
  })
})