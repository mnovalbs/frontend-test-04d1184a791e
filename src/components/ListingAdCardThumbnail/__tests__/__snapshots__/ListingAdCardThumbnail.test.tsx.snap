// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingAdCardThumbnail renders thumbnail 1`] = `
<div>
  <div
    class="thumbnailContainer"
  >
    <img
      alt="Forett At Bukit Timah"
      class="thumbnail"
      height="272"
      loading="lazy"
      src="https://images.prismic.io/99-content/dc1594fb-f413-44ff-b8b5-c3ba6cd539cd_vbWkNuxt2hjgidVttoMRDX.jpg?mode=crop&auto=compress&w=544&q=35"
      srcset="https://images.prismic.io/99-content/dc1594fb-f413-44ff-b8b5-c3ba6cd539cd_vbWkNuxt2hjgidVttoMRDX.jpg?mode=crop&auto=compress&w=544&q=35, https://images.prismic.io/99-content/dc1594fb-f413-44ff-b8b5-c3ba6cd539cd_vbWkNuxt2hjgidVttoMRDX.jpg?mode=crop&auto=compress&w=1088&q=25 2x"
      width="544"
    />
  </div>
</div>
`;
